apps/prairielearn/src/pages/instructorAssessmentGroups/instructorAssessmentGroups.js

```diff
@@ -104,43 +103,33 @@ router.post(
     } else if (req.body.__action === 'add_group') {
       const assessment_id = res.locals.assessment.id;
       const group_name = req.body.group_name;
-      if (!group_name || String(group_name).length < 1) {
-        flash('error', 'Group name cannot be empty.');
-        res.redirect(req.originalUrl);
-        return;
-      }
 
       const uids = req.body.uids;
       const uidlist = uids.split(/[ ,]+/).filter((uid) => !!uid);
-      if (uidlist.length === 0) {
-        flash('error', 'Group must be created with at least one user.');
-        res.redirect(req.originalUrl);
-        return;
+      const userIdList = [];
+      const notExist = [];
+      for (const uid of uidlist) {
+        const user = await selectUserByUid(uid);
+        if (user == null) {
+          notExist.push(uid);
+        } else {
+          userIdList.push(user.user_id);
+        }
```

Comment:
While I appreciate the elimination of a sproc, I don't like that this essentially turns into an N+1 query, _especially_ since we don't bound the number of users someone could add to a group.

Reply:
This was removed from here, but it still checks for UID/enrolment inside the transaction. Is this ok?

Reply:
I can live with that!